### PR TITLE
doc: Add interop deployment documentation

### PR DESCRIPTION
**Pull request by michalborkowski96**
#### Summary
https://github.com/TheThingsIndustries/lorawan-stack-aws/pull/505

Interop is going to actually work from version 3.16.0

#### Changes
Added interop documentation

#### Notes for Reviewers
@johanstokking can you go through it and fix issues?
@benolayinka can you help me with style?

#### Checklist

- [X] Scope: The referenced issue is addressed, there are no unrelated changes.
- [X] Run Locally: Verified that the docs build using `make server`, posted screenshots, verified external links. Test with `HUGO_PARAMS_SEARCH_ENABLED=true` if style changes will affect the search bar.
- [ ] New Features Marked: Documentation for new features is marked using the `new-in-version` shortcode, according to the guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [ ] Style Guidelines: Documentation obeys style guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [X] Commits: Commit messages follow guidelines in [CONTRIBUTING](CONTRIBUTING.md), there are no fixup commits left.

### CODE REVIEW

File: doc/content/getting-started/aws/ecs/interop/_index.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+---
+title: "Interoperability"
+description: "LoRaWAN Backend Interfaces"
+weight: 3
+---
+
+{{% tts %}} offers support for LoRaWAN Backend Interfaces. Interoperability can be turned on on JoinServer or IdentityServer, but not both at once, as they use the same port.
```
Comment (johanstokking) [inline doc/content/getting-started/aws/ecs/interop/_index.md:7]:
```suggestion
{{% tts %}} exposes the Join Server API defined by LoRaWAN Backend Interfaces 1.0 and 1.1. This is used for interoperability between LoRaWAN networks. The functionality can be exposed either by the Join Server or by the Identity Server, but not both at once.
```

File: doc/content/getting-started/aws/ecs/interop/_index.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+# JoinServer interoperability
```
Comment (johanstokking) [inline doc/content/getting-started/aws/ecs/interop/_index.md:9]:
```suggestion
# Join Server interoperability
```

File: doc/content/getting-started/aws/ecs/interop/_index.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+## Architecture
+
+Support for LoRaWAN Backend Interfaces requires certain extensions to the typical AWS ECS deployment architecture. First, we need to include CryptoServer in the deployment. This component is responsible for cryptographic operations in the context of Microchip secure element. Since it needs access to highly sensitive data, {{% tts %}} assumes that it's not deployed together with the rest of the stack, but in a separate account. In this way regular ops team can have access to the most part of the stack, while CryptoServer, together with highly sensitive Microchip secure element encryption keys, would be accessed only by selected members of your organization.
+
+In order to provide this, {{% tts %}} uses VPC peering functionality of AWS.
```
Comment (johanstokking) [inline doc/content/getting-started/aws/ecs/interop/_index.md:15]: Remove; this is not relevant for public documentation. Also, how things are operated in TTSC(E) are not documented publicly.

File: doc/content/getting-started/aws/ecs/interop/_index.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+Moreover, JoinServer needs to validate the client certificate when establishing TLS connections. This means that, while the rest of the stack has its TLS terminated by the AWS Load Balancer, JoinServer needs access to the TLS certificate of your server and of connecting clients.
```
Comment (johanstokking) [inline doc/content/getting-started/aws/ecs/interop/_index.md:17]: Idem; too deployment specific

**Review by johanstokking — CHANGES_REQUESTED** [review]
This should not be documented here

If anything, do document that Backend Interfaces is exposed by JS or IS, and what that means. In short, JS can handle the join flow (answer to `HomeNSReq`, `JoinReq`, `AppSKeyReq`), while IS can only answer to `HomeNSReq`. People should typically enable IS or JS if they want to use device activations through Packet Broker, and expose JS if they want to expose the entire join flow to other LoRaWAN network servers.

File: doc/content/getting-started/aws/ecs/interop/_index.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+Join Server is capable of handling the join flow, that is `HomeNSReq`, `JoinReq` and `AppSKeyReq` requests. Identity Server can only answer to `HomeNSReq`. Typically, you should enable interoperability if you want to use device activations through Packet Broker. You can use Join Server interoperability if you want to expose the entire join flow to other LoRaWAN network servers.
+
+# Join Server interoperability
+
+## Deployment
+
+### Step 1: deploy {{% tts %}}
+
+Deploy {{% tts %}} normally. For the `PeerRequesterAccountId` parameter in `1-1-vpc`, use the ID of the account where you want to deploy Crypto Server. Do not enable interop and assume that {{% tts %}} doesn't connect to the Crypto Server.
```
Comment (johanstokking) [inline doc/content/getting-started/aws/ecs/interop/_index.md:17]: There is no need to document anything about Crypto Server at this point. It is part of TTSE but it is unusable by customers in its current shape and form.

File: doc/content/getting-started/aws/ecs/interop/_index.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+### Step 2: add certificates
+
+For the server certificate, use the certbot task. It will request a certificate and put it into AWS Secrets Manager. If you do not want to use the certbot task, but prefer to provide your own certificate, look up the relevant secret created in template `4-1-secrets` and substitute the dummy value. Certificates are expected to be in PEM format.
+
+Client certificates are stored in the interop config bucket, created in template `2-4b-routing-s3`. In this bucket you need to place a file `config.yml`, which contains a dictionary `NetID -> filename_with_certificate_in_pem_format`. Example configuration:
```
Comment (johanstokking) [inline doc/content/getting-started/aws/ecs/interop/_index.md:23]:
```suggestion
Client certificates are stored in the interop config bucket, created in template `2-4b-routing-s3`. In this bucket you need to place a file `config.yml`, which contains a map `<net-id>: <ca.pem>`. Example configuration:
```

File: doc/content/getting-started/aws/ecs/interop/_index.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+```
```
Comment (johanstokking) [inline doc/content/getting-started/aws/ecs/interop/_index.md:25]:
```suggestion
```yaml
```

File: doc/content/getting-started/aws/ecs/interop/_index.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+2. `aws kms encrypt --cli-input-json file://input.json`. It is important to use the input json file so that secret value doesn't stay in history
+3. Change the contents of `input.json` to `{"TableName": "<MicrochipPartsTableKey>", "Item": {"part_number": {"S": "<PartNumber>"}, "psk_root_keys_encrypted": {"S": "<CiphertextBlobFromKMSEncrypt>"}}}`
+4. `aws dynamodb put-item --cli-input-json file://input.json`
+5. `rm input.json`
+
+At this point interoperability should be working. It might be required to restart the containers before they pick up new configuration.
+
+# Identity Server interoperability
+
+Identity Server interoperability doesn't need Crypto Server, therefore you don't need to set up two accounts and peer networks. Still, you need to set up TLS certificates in a similar fasion.
```
Comment (johanstokking) [inline doc/content/getting-started/aws/ecs/interop/_index.md:67]:
```suggestion
Identity Server interoperability doesn't need Crypto Server, therefore you don't need to set up two accounts and peer networks. Still, you need to set up TLS certificates in a similar fashion.
```

File: doc/content/getting-started/aws/ecs/interop/_index.md (hunk 5 above)
Comment (michalborkowski96) [reply doc/content/getting-started/aws/ecs/interop/_index.md:17]: I think we should document this. This documentation isn't only for the customers, but also for ourselves. I mean, you spent hours looking up how Global Join Server actually works, so when we're doing a new setup, it's worth it to document it, so we don't run into the same problem again in the future.

Also, without the Crypto Server part, the documentation boils down to "TTS supports interop", so I can simply close this PR without merging.

Comment (johanstokking) [reply doc/content/getting-started/aws/ecs/interop/_index.md:17]: OK

**Review by johanstokking — APPROVED** [review]

**Pull request merged**